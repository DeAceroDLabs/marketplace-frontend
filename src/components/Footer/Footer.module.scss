@use "styles/colors";
@use 'styles/typography';

.footer {
  margin-top: 20px;
  background-color: colors.$color-background-03;
  height: 250px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  .info-white-container {
    background-color: colors.$color-background-02;
    height: 170px;
    width: 100%;
    padding: 20px;
  }

  .info-text {
    display: flex;
    justify-content: space-around;
    width: 40%;
    margin-top: 15px;
    margin-left: 3%;
  }
  .popup{
    left: 40%;
    bottom: 20px;
  }

  .popup-text {
    width: 374px;
    height: 110px;
    background-color: red;
    margin-left: 20px;

    .copy-email {
      color: colors.$accent;
      font-size: typography.$body-m-size;
    }

  }

}