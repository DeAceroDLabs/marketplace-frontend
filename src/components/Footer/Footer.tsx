import { useNavigate, Link } from "react-router-dom";
import { useContext, useState } from "react";
import UserContext from "config/userContext";
import Button from "components/common/Button";
import styles from "./Footer.module.scss";
import Tooltip from "components/common/Tooltip";
import Section from "components/common/Section";

const Footer: React.FC = () => {
  const navigate = useNavigate();
  const { username } = useContext(UserContext);

  const [isOpen, setOpen] = useState(false);
  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  if (!username) {
    return null;
  }
  const tooltipContent = (
    <div className={styles["tooltip-text"]}>
      <div className={styles.title}>Soporte</div>
      <p>Envianos tus dudas a: </p>
      <Link
        className={styles["copy-email"]}
        to="#"
        onClick={(e) => {
          window.location.href = "ejemplo@correo.com";
          e.preventDefault();
        }}
      >
        {"ejemplo@correo.com"}
      </Link>
    </div>
  );

  const contactSupport = (
    <div>
      <Button color="transparent-black" action={() => void 0}>
        Contactar soporte
      </Button>
    </div>
  );

  return (
    <div className={styles.footer}>
      <div className={styles["info-white-container"]}>
        <div className={styles["info-text"]}>
          <Section title="Acerca de" variant="no-margin-title">
            <div className={styles.buttons}>
              <Button
                color="transparent-black"
                action={() => navigate("/conocenos")}
              >
                Con√≥cenos
              </Button>
              <Button
                color="transparent-black"
                action={() => navigate("/ayuda")}
              >
                Ayuda
              </Button>
            </div>
          </Section>
          <Section title="Contacto" variant="no-margin-title">
            <div className={styles.buttons}>
              <Tooltip
                offset={[4, 0]}
                position="right center"
                triggerTooltip={contactSupport}
                isOpen={isOpen}
                handleClose={handleClose}
                handleOpen={handleOpen}
              >
                {tooltipContent}
              </Tooltip>
              <Button
                color="transparent-black"
                action={() => navigate("/preguntas")}
              >
                Preguntas frecuentes
              </Button>
              <Button
                color="transparent-black"
                action={() => navigate("/politicas")}
              >
                Politicas y condiciones
              </Button>
            </div>
          </Section>
        </div>
      </div>
    </div>
  );
};

export default Footer;
