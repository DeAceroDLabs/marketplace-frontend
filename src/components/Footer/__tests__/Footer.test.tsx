import { render, screen, fireEvent,waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Footer from "../Footer";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Footer", () => {
  it("renders Footer without problem", async() => {
    const view = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });

});