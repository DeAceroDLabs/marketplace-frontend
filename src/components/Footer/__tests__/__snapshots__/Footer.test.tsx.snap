// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer without problem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer"
      >
        <div
          class="subscribing"
        >
          <div
            class="text"
          >
            Sé el primero en enterarte de ofertas y novedades. Compártenos tu correo y te mantendremos al tanto de todo.
          </div>
          <div
            class="email"
          >
            <input
              placeholder="Escribe tu correo electónico"
              type="text"
            />
            <div
              class="button-container"
            >
              <button
                class="button suscribe"
              >
                Suscribete
              </button>
            </div>
          </div>
        </div>
        <div
          class="info-white-container"
        >
          <div
            class="info-text"
          >
            <div
              class="acerca-de"
            >
              <b>
                Acerca de
              </b>
              <button
                class="button transparent-black"
              >
                Conócenos
              </button>
              <button
                class="button transparent-black"
              >
                Ayuda
              </button>
            </div>
            <div
              class="contacto"
            >
              <b>
                Contacto
              </b>
              <button
                class="button transparent-black"
              >
                Contactar soporte
              </button>
              <button
                class="button transparent-black"
              >
                Preguntas frecuentes
              </button>
              <button
                class="button transparent-black"
              >
                Politicas y condiciones
              </button>
            </div>
            <div
              class="popup"
              id="myPopup"
            >
              <b>
                Soporte
              </b>
              <div
                class="popup-text"
              >
                <p>
                  Envianos tus dudas a: 
                </p>
                <p>
                  ejemplo@correo.com 
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer"
    >
      <div
        class="subscribing"
      >
        <div
          class="text"
        >
          Sé el primero en enterarte de ofertas y novedades. Compártenos tu correo y te mantendremos al tanto de todo.
        </div>
        <div
          class="email"
        >
          <input
            placeholder="Escribe tu correo electónico"
            type="text"
          />
          <div
            class="button-container"
          >
            <button
              class="button suscribe"
            >
              Suscribete
            </button>
          </div>
        </div>
      </div>
      <div
        class="info-white-container"
      >
        <div
          class="info-text"
        >
          <div
            class="acerca-de"
          >
            <b>
              Acerca de
            </b>
            <button
              class="button transparent-black"
            >
              Conócenos
            </button>
            <button
              class="button transparent-black"
            >
              Ayuda
            </button>
          </div>
          <div
            class="contacto"
          >
            <b>
              Contacto
            </b>
            <button
              class="button transparent-black"
            >
              Contactar soporte
            </button>
            <button
              class="button transparent-black"
            >
              Preguntas frecuentes
            </button>
            <button
              class="button transparent-black"
            >
              Politicas y condiciones
            </button>
          </div>
          <div
            class="popup"
            id="myPopup"
          >
            <b>
              Soporte
            </b>
            <div
              class="popup-text"
            >
              <p>
                Envianos tus dudas a: 
              </p>
              <p>
                ejemplo@correo.com 
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
